Lib/test/test_asyncio/test_eager_task_factory.py

```diff
@@ -0,0 +1,241 @@
+        self.loop = asyncio.new_event_loop()
+        self.loop.set_task_factory(asyncio.eager_task_factory)
+        self.set_event_loop(self.loop)
+
+    def test_eager_task_factory_set(self):
+        self.assertIs(self.loop.get_task_factory(), asyncio.eager_task_factory)
+
+    def test_await_future_during_eager_step(self):
+
+        async def set_result(fut, val):
+            fut.set_result(val)
+
+        async def run():
+            fut = self.loop.create_future()
+            t = self.loop.create_task(set_result(fut, 'my message'))
+            # assert the eager step completed the task
+            self.assertTrue(t.done())
+            return await fut
+
+        result = self.loop.run_until_complete(run())
+        self.assertEqual(result,'my message')
+
+    def test_eager_completion(self):
+
+        async def coro():
+            return 'hello'
+
+        async def run():
+            await asyncio.sleep(0.1)
+            t = self.loop.create_task(coro())
+            # assert the eager step completed the task
+            self.assertTrue(t.done())
+            return await t
+
+        result = self.loop.run_until_complete(run())
+        self.assertEqual(result, 'hello')
+
+    def test_block_after_eager_step(self):
+
+        async def coro():
+            await asyncio.sleep(0.1)
+            return 'finished after blocking'
+
+        async def run():
+            await asyncio.sleep(0.1)
+            t = self.loop.create_task(coro())
+            self.assertFalse(t.done())
+            result = await t
+            self.assertTrue(t.done())
+            return result
+
+        result = self.loop.run_until_complete(run())
+        self.assertEqual(result, 'finished after blocking')
+
+    def test_cancellation_after_eager_completion(self):
+
+        async def coro():
+            return 'finished without blocking'
+
+        async def run():
+            await asyncio.sleep(0.1)
+            t = self.loop.create_task(coro())
+            t.cancel()
+            result = await t
+            # finished task can't be cancelled
+            self.assertFalse(t.cancelled())
+            return result
+
+        result = self.loop.run_until_complete(run())
+        self.assertEqual(result, 'finished without blocking')
+
+    def test_cancellation_after_eager_step_blocks(self):
+
+        async def coro():
+            await asyncio.sleep(0.1)
+            return 'finished after blocking'
+
+        async def run():
+            await asyncio.sleep(0.1)
+            t = self.loop.create_task(coro())
+            t.cancel('cancellation message')
+            self.assertGreater(t.cancelling(), 0)
+            try:
+                result = await t
+            except asyncio.CancelledError:
+                # finished task can't be cancelled
+                self.assertTrue(t.cancelled())
+                raise
+
+        with self.assertRaises(asyncio.CancelledError) as cm:
+            self.loop.run_until_complete(run())
```

Comment:
I don't think we need to re-raise + catch the `CancelledError`?

```suggestion
            with self.assertRaises(asyncio.CancelledError):
                result = await t
            self.assertTrue(t.cancelled())

        self.loop.run_until_complete(run())
```